### PR TITLE
Correctly mark validations as full or partial

### PR DESCRIPTION
**Pull request by wilsonianb**
* Correctly set kFullFlag in STValidation constructor
* Include validation type (full or partial) in subscription stream

### CODE REVIEW

**Comment by vinniefalco** [comment]
Suggest: Append a colon to the end of the first line of the commit message, and make sure the second line of the commit message is a blank line.

**Comment by vinniefalco** [comment]
Changes look good
:+1:

File: src/ripple/app/misc/NetworkOPs.cpp [hunk 1]
```diff
@@ -1623,6 +1623,7 @@ void NetworkOPsImp::pubValidation (STValidation::ref val)
             val->getSignerPublic());
         jvObj [jss::ledger_hash]           = to_string (val->getLedgerHash ());
         jvObj [jss::signature]             = strHex (val->getSignature ());
+        jvObj [jss::validation_type]       = val->isFull() ? "full" : "partial";
```
Comment (miguelportilla) [inline src/ripple/app/misc/NetworkOPs.cpp]: Suggest using jss::full and define jss::partial

**Comment by miguelportilla** [comment]
Left suggestion, not holding back. :+1:

File: src/ripple/app/misc/NetworkOPs.cpp (hunk 1 above)
Comment (wilsonianb) [inline src/ripple/app/misc/NetworkOPs.cpp]: Would the message always include both `full` and `partial` with one marked `true` and the other `false`?

Comment (wilsonianb) [inline src/ripple/app/misc/NetworkOPs.cpp]: I considered using `jss::full` and marking it `true` or `false`

Comment (vinniefalco) [inline src/ripple/app/misc/NetworkOPs.cpp]: @miguelportilla Can you please give a code example for your suggestion?

**Comment by ximinez** [comment]
The changes look good, but I'm concerned that reversing the flag set logic (even though it's a correction) will have unintended consequences. I can't _find_ any, but I thought I'd mention it.

**Comment by nbougalis** [comment]
The changes look good. I echo @ximinez's concern re: flag inversion.

**Comment by ximinez** [comment]
I did a little more digging, and as far as I can find, there are 3 calls to `STValidation::isFull`

1. In `NetworkOPsImp::pubValidation`. New, added by this PR.
2. In `ValidationsImp::getValidationTypes`. Not referenced / called.
3. In `ValidationsImp::getLoadRatio`. Not reference / called.

So I guess this change is safe after all, at least within the scope of `rippled`. Might not hurt to remove those two unused functions.

File: src/ripple/app/misc/NetworkOPs.cpp (hunk 1 above)
Comment (miguelportilla) [inline src/ripple/app/misc/NetworkOPs.cpp]: I meant:

jvObj [jss::validation_type] = val->isFull() ? jss::full : jss::partial;

Comment (miguelportilla) [inline src/ripple/app/misc/NetworkOPs.cpp]: @wilsonianb I like your idea of jss::full and marking it true or false. Claiming 'partial' seems misleading to me.

Comment (nbougalis) [inline src/ripple/app/misc/NetworkOPs.cpp]: I'd use `jvObj[jss::full] = val->isFull();`

**Comment by miguelportilla** [comment]
@ximinez The correction Brandon made to the ctor seems correct. As you mentioned, isFull is not used otherwise Validations may not have behaved as intended.

**Comment by nbougalis** [comment]
:+1:

**Pull request closed without merging**